src/coreclr/vm/inlinetracking.cpp

```diff
@@ -424,7 +424,235 @@ Module* PersistentInlineTrackingMapR2R2::GetModuleByIndex(DWORD index)
+                streamSize--;
+                inlineeModule = GetModuleByIndex(moduleIndex);
+            }
+            else
+            {
+                inlineeModule = m_module;
+            }
+        }
+
+        if (inlineeModule != inlineeOwnerMod)
+        {
+            continue;
+        }
+
+        // We have the right inlinee, let's look at the inliners
+
+        // First chack for cross module inliners
+        if (hasCrossModuleInliners)
+        {
+            _ASSERTE(streamSize > 0);
+            uint32_t crossModuleInlinerCount = entryParser.GetUnsigned();
+            streamSize--;
+
+            for (uint32_t iInliner = 0; iInliner < crossModuleInlinerCount; iInliner++)
+            {
+                _ASSERTE(streamSize > 0);
+                uint32_t inlinerIndex = entryParser.GetUnsigned();
+                streamSize--;
+
+                Module* inlinerModule = NULL;
+                mdMethodDef inlinerToken = 0;
+                GetILBodyTokenInfo(inlinerIndex, ppMethods, cMethods, &inlinerModule, &inlinerToken);
+                AddInliner(inlinersSize, inliners, &result, inlinerModule, inlinerToken, incompleteData);
+            }
+        }
+
+        // Then handle the non-cross module ones
+        DWORD currentInlinerIndex = 0;
+        while (streamSize > 0)
+        {
+            uint32_t inlinerIndexDeltaAndFlag = entryParser.GetUnsigned();
+            streamSize--;
+
+            Module* inlinerModule = inlineeModule;
+
+            if (multiModuleFormat)
+            {
+                currentInlinerIndex += inlinerIndexDeltaAndFlag >> (int)ReadyToRunCrossModuleInlineFlags::InlinerRidShift;
+                if ((((ReadyToRunCrossModuleInlineFlags)inlinerIndexDeltaAndFlag) & ReadyToRunCrossModuleInlineFlags::InlinerRidHasModule) == ReadyToRunCrossModuleInlineFlags::InlinerRidHasModule)
+                {
+                    _ASSERTE(streamSize >= 0);
+                    uint32_t moduleIndex = entryParser.GetUnsigned();
+                    streamSize--;
+                    inlinerModule = GetModuleByIndex(moduleIndex);
+                }
+            }
+            mdMethodDef inlinerToken = TokenFromRid(currentInlinerIndex, mdtMethodDef);
+
+
+            AddInliner(inlinersSize, inliners, &result, inlinerModule, inlinerToken, incompleteData);
+        }
+    }
+
+    return result;
+}
+
+Module* CrossModulePersistentInlineTrackingMapR2R::GetModuleByIndex(DWORD index)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    // This "black magic spell" has in fact nothing to do with GenericInstantiationCompare per se, but just sets a thread flag
+    // that later activates more thorough search inside Module::GetAssemblyIfLoaded, which is indirectly called from GetModuleFromIndexIfLoaded.
+    // This is useful when ngen image was compiler against a different assembly version than the one loaded now.
```

Comment:
Nit - typo (compiler -> compiled). Also, do we still want to refer to R2R images as "ngen"?